### ISSUE TITLE
Add ELB/ALB support

### ISSUE THREAD

**Issue by miere** [localstack/localstack#25]
Hi atlassian developers

I'd like to suggest you guys to include AWS Elastic (and Application) Load Balancing support. It would be nice to have a reliable implementation of these AWS's services to use on our unit tests.

┆Issue is synchronized with this [Jira Bug](https://localstack.atlassian.net/browse/LOC-194) by [Unito](https://www.unito.io/learn-more)

**Comment by whummer** [comment]
Thanks for the suggestion, we'll look into it.

A potential problem is that there are some inherent limitations due to the fact that we're executing things locally. For instance, the ELB `CreateLoadBalancer` API method returns a DNS name, which will require some adjustments to the tests if they are executed locally: http://docs.aws.amazon.com/elasticloadbalancing/2012-06-01/APIReference/API_CreateLoadBalancer.html#API_CreateLoadBalancer_ResponseElements
Potential solutions that come to mind would be to 1) have the tests send requests to `localhost` and send along a `Host: ...` HTTP header with the DNS name which allows the ELB mock to look up the corresponding backend service, or 2) actually update the `/etc/hosts` configuration file on the client (which typically requires root access and is therefore not really an option).

Another aspect to consider is that, if we want to support `LocalStack` ELB in Docker mode, we need to find a reliable way to forward any invocations hitting the ELB mock *inside* the Docker container to target services running *outside* the container (on the host operating system, or potentially in another Docker container running in parallel).

So there are a couple of things to consider if we want to add generic support for ELB/ALB. Did you have any particular use case in mind? Would be great to have a starting point, maybe an existing unit/integration test?

**Comment by nilathj** [comment]
I would like the ability to trigger a lambda from an ALB end point.

**Comment by SJ-SivaSubrahmaniam** [comment]
We are facing issues, having Node multiple instances/ALB routing, If we  had this setup in local, it will be better

**Comment by terozio** [comment]
Old feature request but this would be really beneficial to have!

We would also like to be able use ALB+Lambda setup locally, could definitely go for the paid version if this was made available.

**Comment by kishaningithub** [comment]
it would be great if we have target groups and operations like "DescribeTargetHealth"

**Comment by subratamazumder** [comment]
@terozio I am working on small POC to mock [ALM-Lambda](https://github.com/subratamazumder/local-alb-lambda) in local. It's at a very initial state using python. Might be of your interest.

**Comment by liubog2008** [comment]
How about this feature?

**Comment by grinsteindavid** [comment]
hi, without elb.. is very complicate to test a simple aws stack

**Comment by monil1334** [comment]
@silv-io any update with supporting ALB in localstack? Do we have a timeline to support this feature?

**Comment by silv-io** [comment]
We're happy to inform you that LocalStack already supports CRUD operations and basic routing for ELB in our pro version. However, if you require any specific features that are not currently supported, we encourage you to create a new issue on our Github repository with your requirements. Our team will review your request and try to incorporate it into LocalStack in the future.
